pkgs/games/spring/default.nix

```diff
@@ -68,41 +56,37 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ cmake makeWrapper docbook_xsl docbook_xsl_ns asciidoc ];
   buildInputs = [
-    xz
     boost
+    curl
+    freetype
+    glew
     libdevil
-    zlib
-    p7zip
-    openal
+    libGL
+    libGLU
+    libunwind
     libvorbis
-    freetype
+    minizip
+    openal
+    p7zip
     SDL2
     xorg.libX11
     xorg.libXcursor
-    libGLU
-    libGL
-    glew
-    curl
-    systemd
-    libunwind
-    which
-    minizip
+    xz
+    zlib
   ]
-  ++ lib.optional withAI jdk
-  ++ lib.optional withAI python;
-
-  NIX_CFLAGS_COMPILE = "-fpermissive"; # GL header minor incompatibility
+  ++ lib.optionals withAI [ python3 jdk ];
 
   postInstall = ''
     wrapProgram "$out/bin/spring" \
-      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ stdenv.cc.cc systemd ]}"
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ stdenv.cc.cc ]}"
   '';
 
   meta = with lib; {
     homepage = "https://springrts.com/";
     description = "A powerful real-time strategy (RTS) game engine";
     license = licenses.gpl2Plus;
     maintainers = with maintainers; [ qknight domenkozar sorki ];
-    platforms = [ "i686-linux" "x86_64-linux" ];
+    platforms = platforms.unix;
+    broken = with stdenv.hostPlatform; !isx86 || isDarwin;
```

Comment:
```suggestion
    platforms = [ "aarch64-linux" "x86_64-linux" ];
```

Reply:
The game does support other platforms though, and I was hoping that someone with Darwin or ARM expertise would fix this later.

But ok, I will revert back on how it was (BTW, this does not build in `aarch64-linux` yet).

Reply:
Actually, upstream removed support for `i686-linux` in the latest release. Will just leave `x86_64-linux` for now.